### ISSUE TITLE
Unable to use save derived rules

### ISSUE THREAD

**Issue by dsanson** [Carnap/Carnap#64]
A student just came to my office with an odd problem. He had proven several derived rules, and they showed up in his list of derived rules. But when he tried to use a rule (on rules page, in the propositional rule builder) he got the "looks like you are citing a derived rule that doesn't exist" error. He says that this error had persisted through logging in and logging out, and quitting and reopening his browser. While I watched, the error persisted after reloading the page. I then asked him to log in and log out, which he did, and the error went away, allowing him to use the derived rules again.

As he was logging in and logging out, I tried the same thing on my laptop, which was already logged in, and had the same error. I logged out and logged back in, and the error went away for me too.

It is as though we were logged in for the purposes of displaying our rule list, but were not logged in as far as the derivation checker was concerned? Is this a known bug?

**Comment by gleachkr** [comment]
No, I don't think I've seen this before. I don't seem to be able to reproduce it right now, but the table of rules and the rules themselves are retrieved in two separate requests to the server, along different routes, so it's conceivable you could have something (network issue, stale cookie) going wrong with one and not the other. I don't know why it would be a persistant issue though (maybe something to do with browser cacheing?). I'll keep an eye out for similar reports. If you can figure out how to reproduce, let me know!
